libs/ml/include/ml/optimisation/optimiser.hpp

```diff
@@ -46,7 +46,7 @@ class Optimiser
             DataType const &learning_rate = DataType{0.001f});
 
   // TODO (private 1090): Optimise TensorSlice for graph-feeding without using .Copy
-  DataType DoBatch(ArrayType &data, ArrayType &labels);
+  DataType DoBatch(ArrayType &data, ArrayType &labels, SizeType batch_size = 0);
```

Comment:
I think this method can be renamed now that it's ALWAYS batchwise; i.e. you can just rename this to "Run" right?